sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveUDFSuite.scala

```diff
@@ -487,6 +488,52 @@ class HiveUDFSuite extends QueryTest with TestHiveSingleton with SQLTestUtils {
     assert(count4 == 1)
     sql("DROP TABLE parquet_tmp")
   }
+
+  test("Hive Stateful UDF") {
+    withUserDefinedFunction("statefulUDF" -> true, "statelessUDF" -> true) {
+      sql(s"CREATE TEMPORARY FUNCTION statefulUDF AS '${classOf[StatefulUDF].getName}'")
+      sql(s"CREATE TEMPORARY FUNCTION statelessUDF AS '${classOf[StatelessUDF].getName}'")
+      withTempView("inputTable") {
+        val testData = spark.sparkContext.parallelize(
+          (0 until 10) map (x => IntegerCaseClass(1)), 2).toDF()
+        testData.createOrReplaceTempView("inputTable")
+        // Distribute all rows to one partition (all rows have the same content),
```

Comment:
@cloud-fan  Thanks for the review. Because all rows only contains IntegerCaseClass(1), RepartitionByExpression will assign all rows to one partition, which has 10 records.